### PR TITLE
Pv lookup

### PR DESCRIPTION
**Pull request by tasleson**
Handle symlinks for PV device paths.

### CODE REVIEW

**Comment by tasleson** [comment]
@vpodzime Please give this a review!

File: test/dbus/lvmdbustest.py [hunk 1]
```diff
@@ -1865,6 +1867,46 @@ def test_nesting(self):
 		for i in range(0, 5):
 			pv_object_path = self._create_nested(pv_object_path)
 
+	def test_pv_symlinks(self):
+		# Lets take one of our test PVs, pvremove it, find a symlink to it
+		# and re-create using the symlink to ensure we return an object
+		# path to it.  Additionally, we will take the symlink and do a lookup
+		# (Manager.LookUpByLvmId) using it and the original device path to
+		# ensure that we can find the PV.
+		symlink = ""
```
Comment (vpodzime) [inline test/dbus/lvmdbustest.py]: I think ``None`` is a better value here.

File: test/dbus/lvmdbustest.py [hunk 2]
```diff
@@ -1865,6 +1867,46 @@ def test_nesting(self):
...
+		self.assertEqual(rc, '/')
+
+		# Lets locate a symlink for it
+		devices = glob('/dev/disk/*/*')
+		for d in devices:
+			if pv_device_path == os.path.realpath(d):
+				symlink = d
+				break
+		else:
+			self.assertTrue(False, "We expected to find at least 1 symlink!")
```
Comment (vpodzime) [inline test/dbus/lvmdbustest.py]: ``self.assertIsNotNone(symlink, ...)`` after the ``for`` cycle would be better here. And the ``for`` cycle could be replaced with a ``next()`` call using a generator expression and ``None`` as the default value. But I can imagine the ``for`` cycle can be better readable.

**Review by vpodzime — APPROVED** [review]
Looks good to me otherwise.

**Comment by tasleson** [comment]
@vpodzime  Thanks for the review, I've updated the code to reflect your suggestions!

File: test/dbus/lvmdbustest.py [hunk 3]
```diff
@@ -1865,6 +1867,47 @@ def test_nesting(self):
...
+		self.assertEqual(rc, '/')
+
+		# Lets locate a symlink for it
+		devices = glob('/dev/disk/*/*')
+		for d in devices:
+			if pv_device_path == os.path.realpath(d):
+				symlink = d
+				break
+		else:
+			self.assertIsNotNone(symlink,
```
Comment (vpodzime) [inline test/dbus/lvmdbustest.py]: I think this should just go after the cycle not to the ``else:`` branch.

Comment (tasleson) [reply test/dbus/lvmdbustest.py]: @vpodzime Well technically the else branch is only executed if a `break` isn't encountered in the for loop, but I get your point and I will remove it.

**Comment by tasleson** [comment]
Changes pushed to upstream

**Pull request closed without merging**